src/collection.js

```diff
@@ -114,6 +127,48 @@ export default class Collection {
   }
 
   /**
+   * Validates an idSchema.
+   *
+   * @param  {Object} idSchema
+   * @return {Object}
+   */
+  _validateIdSchema(idSchema) {
+    if (!idSchema) {
+      return createUUIDSchema();
+    }
+    if (typeof idSchema !== "object") {
+      throw new Error("idSchema must be an object.");
+    } else if (typeof idSchema.generate !== "function") {
+      throw new Error("idSchema must provide a generate function.");
+    } else if (typeof idSchema.validate !== "function") {
+      throw new Error("idSchema must provide a validate function.");
+    }
+    return idSchema;
+  }
+
+  /**
+   * Validates a list of remote transformers.
+   *
+   * @param  {Array} remoteTransformers
+   * @return {Array}
+   */
+  _validateRemoteTransformers(remoteTransformers) {
+    if (!remoteTransformers || remoteTransformers.length === 0) {
```

Comment:
You should check here if remoteTransformers is an error, otherwise this will throw `remoteTransformers.map is not a function` if people pass `{ remoteTransformers: { length: 1 } }` as the collection options. OK, nobody in their right mind would ever pass that of course, so it's all academic, but still. ;)